src/rust/engine/process_execution/src/named_caches.rs

```diff
@@ -0,0 +1,62 @@
+use std::collections::BTreeSet;
+use std::ffi::OsString;
+use std::path::PathBuf;
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash, PartialOrd, Ord)]
+pub struct NamedCache(String);
+
+impl NamedCache {
+  pub fn new(name: String) -> Result<NamedCache, String> {
+    if name.chars().all(|c| c.is_ascii_alphanumeric() || c == '_') {
+      Ok(NamedCache(name.to_lowercase()))
```

Comment:
The lower-casing here is not specified in the docs, so is kind of surprising. I agree that case-canonicalising is probably a good thing, but we should make it clear to the callers, so they know how to construct the correct environment variables.

Also, we should probably document this somewhere outside of the rust code :)

Reply:
Made it explicit.

> Also, we should probably document this somewhere outside of the rust code :)

Yea, I think the whole docstring should probably move to the Python side in a followup.